frontend/src/pages/supervisor/SupervisorDashboard.tsx

```diff
@@ -0,0 +1,98 @@
+import { Loading, Table, TableFooterNav, TableFooterPageSelect } from '@/components';
+import { Transition } from '@headlessui/react';
+import { useSupervisorDashboard } from '@/hooks/useSupervisorDashboard';
+import { SupervisorDashboardHeaderBanner } from './SupervisorDashboardHeader';
+import { BannerType } from '@/common/enums/banner-enum';
+import { Banner } from '@/components/ui/Banner';
+import { useRecommitmentCycle } from '@/hooks/useRecommitment';
+
+const SupervisorDashboard = () => {
+  const {
+    totalRows,
+    currentPage,
+    rowsPerPage,
+    rows,
+    loading,
+    handleChangePage,
+    columns,
+    showSuccessMessage,
+    showWarningBanner,
+    handleCloseSuccessBanner,
+    handleCloseWarningBanner,
+  } = useSupervisorDashboard();
+
+  const recommitmentCycle = useRecommitmentCycle();
+
+  if (loading) return <Loading />;
+
+  return (
+    <div className="max-w-[1150px] w-full h-full flex flex-col items-center justify-start pb-24">
+      <SupervisorDashboardHeaderBanner
+        recommitmentDate={recommitmentCycle?.endDate.toString() ?? ''}
+        recommitmentYear={recommitmentCycle?.year.toString() ?? ''}
+      />
+      <Transition
+        show={showSuccessMessage || showWarningBanner}
+        appear={true}
+        enter="ease-out duration-100"
+        enterFrom="opacity-0 scale-95"
+        enterTo="opacity-100 scale-100"
+        leave="ease-in duration-200"
+        leaveFrom="opacity-100 scale-100"
+        leaveTo="opacity-0 scale-95"
+      >
+        <div className="max-w-[1150px] py-12">
+          {showSuccessMessage && (
+            <Banner
+              onClose={handleCloseSuccessBanner}
+              title={'Supervisor Approval for Recommitment Successfully Submitted'}
```

Comment:
We had talked about lifting out text to some constants, is that something we want to start here?

Reply:
yes, I agree, but would prefer to work on that separately since this PR has grown quite large (if that's cool with you)